### DISCUSSION TITLE
Where to connect TX

### DISCUSSION THREAD

**Discussion by grey1808** [ZZ-Cat/CRSFforArduino#26] [General]
Greetings! Please tell me, maybe you can help me with my one more question, maybe it is not much off topic of your library.
Which pins of the SAMD21 microcontroller to connect RX and TX to, in your library description it says that TX is pin0, and RX is pin1, but there is no pin0 output on my SAMD21 board  and pin1 is not either..
![изображение](https://github.com/ZZ-Cat/CRSFforArduino/assets/44221929/bac56ee4-93ea-485f-8f6e-14970f07dcbf)

**Comment by ZZ-Cat** [comment]
I reverse image searched the pic you linked & it's a [Seeed Studio XIAO](https://www.seeedstudio.com/Seeeduino-XIAO-Arduino-Microcontroller-SAMD21-Cortex-M0+-p-4426.html) with a SAMD21G18A (same MCU as my Feather M0 BlueFruit LE & my two Metro M0 Express boards).
I found the [schematic](https://files.seeedstudio.com/wiki/Seeeduino-XIAO/res/Seeeduino-XIAO-v1.0-SCH-191112.pdf) for it & you have three Sercom ports to play with. The default configuration for your Xiao has UART Rx on D7 & UART Tx on D6. This translates to Sercom 4 pad 1 for Rx & Sercom 4 pad 0 for Tx.

Be aware that your Xiao is not on the compatibility list... yet. So, if you get _this_ message showing up in your serial monitor `[CRSF for Arduino | ERROR] Devboard is not compatible with CRSF for Arduino.` or your `setup()` code stops after `CRSFforArduino::begin()`, then that means your Xiao was flagged as being incompatible. If that happens, open up an [Issue](https://github.com/ZZ-Cat/CRSFforArduino/issues/new/choose) & let me know what's up, yea?

**Pro Tip:** To debug CRSF for Arduino, uncomment `#define CRSF_DEBUG` at line 40 in `CRSFforArduino.h`.

**Reply by grey1808** [reply]
No, I did not observe such an error, thanks to your edit from the previous topic, I was able to compile and upload my program to the board. There was no such error. But thanks for the advice, I'll try to do what you described

**Reply by ZZ-Cat** [reply]
Excellent. =^/,..,^=

Let me know how you get on, yea?

**Reply by grey1808** [reply]
Hello my friend) Nothing worked out for me) Of course, I will unsubscribe to you about my successes, well, if they are of course)
Look, I took your example, and I was able to write it to the board. In theory, I should have seen a message in the console, but there was none.

After the `crsf.begin();` Messages are not output to the console. Before `crsf.begin();` removed, after that it is gone. And nothing is written in the console * I don’t know what to do)
I also tried to receive a signal from the remote control of the copter, but the console still did not respond, which is generally logical, it seems that there is an error somewhere in the library, the error of which I cannot see

**Reply by ZZ-Cat** [reply]
Yea, I think I have a fair idea what's going on there.

IIRC, what is stopping you is _this_ block of code, right here in `CRSFforArduino::begin()`:
```cpp
    /* Check at runtime if the devboard is compatible with CRSF for Arduino. */
    const char *devboardName = CT.getDevboardName();
#ifdef CRSF_DEBUG
    Serial.print("[CRSF for Arduino | INFO] Devboard Name: ");
    Serial.println(devboardName);
#endif

    // Incompatible devboards will be caught here.
    if (CT.isDevboardCompatible(devboardName) != true)
    {
#ifdef CRSF_DEBUG
        Serial.println("[CRSF for Arduino | ERROR] Devboard is not compatible with CRSF for Arduino.");
#endif
        // Stop here, instead of returning false, because CRSF for Arduino is not compatible with this devboard.
        while (1)
            ;
    }
```

By rights, this code above should also trigger a compiler warning in your console along the lines of `Devboard not supported. Please check the compatibility table.` Especially on first compile (if you are using the Arduino IDE).

As I previously stated, I currently don't have your development board on the compatibility list, despite it having a SAM D21 chipset. If this is what I think it is, all I need to do on my end is add your Seed XIAO to the compatibility list, & by rights... it should be all clean sailing for you, from there.

When I wrote out the compatibility list, at the time, I wasn't aware that Seeed Studio still exists. Well, now I know. =^/.~=

PS: The compatibility list also tells CRSF for Arduino what board it's being compiled for, what UART port(s) it should use, & what pins to use. When a board is not compatible, the ports & pins are defined as `null` & will likely not function as intended, if runtime manages to get past your call to `CRSF.begin()` in your main code.

**Comment by ZZ-Cat** [comment]
I have since [answered](https://github.com/ZZ-Cat/CRSFforArduino/discussions/54#discussioncomment-8163189) your question over in #54.  
'Cause someone else with one of the Seeed XIAO targets had a similar problem.
